feathr_project/test/test_azure_spark_e2e.py

```diff
@@ -201,3 +202,73 @@ def test_feathr_get_offline_features():
+                               description="location id in NYC",
+                               full_name="nyc_taxi.location_id")
+
+        feature_query = FeatureQuery(
+            feature_list=["f_location_avg_fare"], key=location_id)
+        settings = ObservationSettings(
+            observation_path="wasbs://public@azurefeathrstorage.blob.core.windows.net/sample_data/green_tripdata_2020-04.csv",
+            event_timestamp_column="lpep_dropoff_datetime",
+            timestamp_format="yyyy-MM-dd HH:mm:ss")
+
+        now = datetime.now()
+        # Set DB user and password before submitting job
+        # os.environ[f"o_USER"] = "some_user@feathrtestsql4"
+        # os.environ[f"o_PASSWORD"] = "some_password"
+        output_path = JdbcSink(name="o",
+                               url="jdbc:sqlserver://feathrtestsql4.database.windows.net:1433;database=testsql;encrypt=true;",
+                               dbtable=''.join(['feathrazure_cijob','_', str(now.minute), '_', str(now.second)]),
+                               auth="USERPASS")
+
+        client.get_offline_features(observation_settings=settings,
+                                    feature_query=feature_query,
+                                    output_path=output_path)
+
+        # assuming the job can successfully run; otherwise it will throw exception
+        client.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
+
+        # We may not have SQL client package installed so skip the verification
+        # res_df = get_result_df(client)
+        # assert res_df.shape[0] > 0
+
+@pytest.mark.skip(reason="We don't have CosmosDb configured in CI pipeline")
+def test_feathr_materialize_to_cosmosdb():
+    """
+    Test FeathrClient() CosmosDbSink.
+    """
+    test_workspace_dir = Path(
+        __file__).parent.resolve() / "test_user_workspace"
+    # os.chdir(test_workspace_dir)
+
+    client: FeathrClient = basic_test_setup(os.path.join(test_workspace_dir, "feathr_config.yaml"))
+
+    backfill_time = BackfillTime(start=datetime(
+        2020, 5, 20), end=datetime(2020, 5, 20), step=timedelta(days=1))
+
+    now = datetime.now()
+    collection = ''.join(['feathrazure_cijob_materialize_cosmosdb_','_', str(now.minute), '_', str(now.second), ""])
+    sink = CosmosDbSink(name='o', endpoint='https://feathrtestcosmosdb.documents.azure.com:443/', database='feathr', collection=collection)
+    settings = MaterializationSettings("nycTaxiTable",
+                                       sinks=[sink],
+                                       feature_names=[
+                                           "f_location_avg_fare", "f_location_max_fare"],
+                                       backfill_time=backfill_time)
+    client.materialize_features(settings)
+    # assuming the job can successfully run; otherwise it will throw exception
+    client.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
```

Comment:
How to verify if the values are actually materialized? We should add command to get those values and check them (for example, not NULL)

Reply:
Added in doc, will update this test later.